Parser/Tree_Nodes/if_node.c

```diff
@@ -0,0 +1,134 @@
+/**************************************************************************************************
+*
+* File Name: if_node.c
+*
+* Author: Jameson Toper
+*
+* Created for CS344 on: 11/06/2019
+*
+* Purpose: Code that provides functions for manipulating If statement nodes.
+*
+**************************************************************************************************/
+
+#include <stdio.h>
+#include "if_node.h"
+#include "b_stmt_list.h"
+
+
+if_node *create_if_node(GArray *token_stream, unsigned long index, unsigned long *next) {
+    if_node *retVal = NULL;
+    retVal = calloc(1, sizeof(if_node));
+    (*next)++;
+    Token *check = &g_array_index(token_stream, Token, *next);
```

Comment:
You should be using index, not next. Next is really only used after you create something with an unknown length. For example, when creating an assigmnment, we create the expression, then use next to make sure the following token is a semicolon